lib/p2p/NodeList.ts

```diff
@@ -96,6 +100,14 @@ class NodeList {
         await node.save();
         return true;
       }
+
+      if (node.lastAddress && addressUtils.areEqual(address, node.lastAddress)) {
+        node.lastAddress = [...node.addresses].sort((a, b) => {
+          if (!a.lastConnected) return 1;
+          if (!b.lastConnected) return -1;
+          return b.lastConnected - a.lastConnected;
+        })[0];
```

Comment:
the sorting logic also exists on `Pool.connectWithAdvertisedAddresses`. We should extract it to `addressUtils`

Reply:
done